llvm/lib/Bitcode/Reader/BitcodeReader.cpp

```diff
@@ -3596,6 +3609,21 @@ Error BitcodeReader::parseConstants() {
                                   Record[1]);
       break;
     }
+    case bitc::CST_CODE_SIGNED_PTR: {
+      if (Record.size() < 6)
+        return error("Invalid record");
+      Type *PtrTy = getTypeByID(Record[0]);
+      if (!PtrTy)
+        return error("Invalid record");
+
+      // PtrTy, Ptr, Key, AddrDiscTy, AddrDisc, Disc
+      V = BitcodeConstant::create(
+        Alloc, CurTy, BitcodeConstant::ConstantPtrAuthOpcode,
+        {(unsigned)Record[1], (unsigned)Record[2], (unsigned)Record[4],
+         (unsigned)Record[5]});
```

Comment:
Why does this skip over `Record[3]`?

Reply:
Okay, I see that this is the addr discriminator type. But does it need to be explicitly encoded, given that you don't actually use it here?

Reply:
Yeah, we don't need either address discriminator type nor base pointer type, I think.  I removed both and added a couple tests and a verifier check for the potentially-relevant cases